### PR TITLE
NxCollapsibleItems RTL Tests - RSC-1235

### PR DESCRIPTION
**Pull request by rpokorny**
https://issues.sonatype.org/browse/RSC-1235

### CODE REVIEW

File: lib/src/components/NxCollapsibleItems/__tests__/NxCollapsibleItems.test.tsx [hunk 1]
```diff
@@ -32,262 +29,261 @@ describe('NxCollapsibleItems', function() {
...
-        <NxCollapsibleItems.Child id="test-id" className="bar" lang="en">foo</NxCollapsibleItems.Child>),
-          div = component.children().children();
+      const minimalProps = { children: 'foo' },
+          quickRender = rtlRender(NxCollapsibleItems.Child, minimalProps),
+          renderEl = rtlRenderElement(NxCollapsibleItems.Child, minimalProps);
 
-      it('renders a div with the listitem role and the nx-collapsible-items__child class', function() {
-        expect(component.hostNodes()).not.toExist();
-        expect(component.children().hostNodes()).not.toExist();
+      it('renders a top-level element with the listitem role', function() {
```
Comment (EmilyBermudez88) [inline lib/src/components/NxCollapsibleItems/__tests__/NxCollapsibleItems.test.tsx:220]: should you also test that the role can be overridden if another role is provided, like in line 182?

File: lib/src/components/NxCollapsibleItems/__tests__/NxCollapsibleItems.test.tsx [hunk 2]
```diff
@@ -32,262 +29,261 @@ describe('NxCollapsibleItems', function() {
...
 
-      expect(component.find('.nx-collapsible-items__trigger')).toHaveProp('aria-controls', childrenElId);
+      const childrenEl = document.getElementById(childrenElId)!,
+          childEl = screen.getByTestId('child');
+
+      expect(view.container).toContainElement(childrenEl);
+      expect(childrenEl).toContainElement(childEl);
     });
 
     it('sets aria-expanded iff both the isOpen prop is true and there are children', function() {
```
[the remark this answers is not in the document]
Comment (rpokorny) [reply lib/src/components/NxCollapsibleItems/__tests__/NxCollapsibleItems.test.tsx:78]: iff means "if and only if"

File: lib/src/components/NxCollapsibleItems/__tests__/NxCollapsibleItems.test.tsx [hunk 3]
```diff
@@ -32,262 +29,261 @@ describe('NxCollapsibleItems', function() {
...
+      await runTimers();
 
-      expect(tooltip).toExist();
-      expect(tooltip).toHaveProp('title', 'tip');
-      expect(tooltip).toHaveProp('placement', 'right');
+      const tooltip = screen.getByRole('tooltip');
 
-      expect(trigger).toExist();
+      expect(tooltip).toHaveTextContent('tip');
+      expect(tooltip.querySelector('.foo')).toBeInTheDocument();
```
Comment (cindyko226) [inline lib/src/components/NxCollapsibleItems/__tests__/NxCollapsibleItems.test.tsx:150]: Is this to test if the classname is set correctly?

Comment (rpokorny) [reply lib/src/components/NxCollapsibleItems/__tests__/NxCollapsibleItems.test.tsx:150]: > Is this to test if the classname is set correctly?

Yes but we don't have control over exactly which element the class gets set on, it's up to how MUI builds the tooltip.  So I can't assert that the class is on the tooltip, but I can assert that it's on some descendant of the tooltip.

Comment (cindyko226) [reply lib/src/components/NxCollapsibleItems/__tests__/NxCollapsibleItems.test.tsx:150]: Make sense, thanks.

File: lib/src/components/NxCollapsibleItems/__tests__/NxCollapsibleItems.test.tsx (hunk 1 above)
Comment (rpokorny) [reply lib/src/components/NxCollapsibleItems/__tests__/NxCollapsibleItems.test.tsx:220]: > should you also test (both here and when the child is an element) that the role can be overridden if another role is provided, like in line 182?

There are actually a lot of nuances around the roles logic here that all deserve to be tested. Good catch. 0c825d96a

**Review by EmilyBermudez88 — APPROVED** [review]
LGTM!

**Pull request merged**